utils/pubsub/pubsub.go

```diff
@@ -98,6 +130,7 @@ type subPublisher struct {
 // internal goroutine if it is not already started
 func (r *subPublisher) publish(data *DataEvent) {
 	r.startedOnce.Do(func() {
+		r.bgCtx = context.Background()
```

Comment:
what is the point of having a background context which cannot be canceled? this case will never materialize with a background context:
```
case <-r.bgCtx.Done():
```